### PR TITLE
build: add devcontainer

### PR DESCRIPTION
**Pull request by damienwebdev**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/develop/CONTRIBUTING.md#commit
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?

```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[ ] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?

Currently, we don't define a [devcontainer](https://code.visualstudio.com/docs/remote/containers) for Daffodil. This also provides a nice integration into https://github.com/features/codespaces. 

Fixes: N/A

## What is the new behavior?

This PR defines a devcontainer, making it easier to get started with development on Daffodil.

Once merged, you can click on the "big green button" in Github and be up and running developing Daffodil just like we do.

Screen Shot 2022-09-01 at 7 24 59 PM

Once we've stabilized this, we can enable prebuilds so that we can get environments started even faster.

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

## Other information

### CODE REVIEW

File: .devcontainer/devcontainer.json [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+		".devcontainer/onCreateCommand.sh"
+	],
+	"extensions": [
+		"stylelint.vscode-stylelint",
+		"naumovs.color-highlight",
+		"bierner.markdown-preview-github-styles",
+		"mrmlnc.vscode-scss",
+		"tehnix.vscode-tidymarkdown",
+		"dbaeumer.vscode-eslint"
+	],
```
Comment (griest024) [inline .devcontainer/devcontainer.json:16]: Why didn't you include `apollographql.vscode-apollo` in this list?

Comment (damienwebdev) [reply .devcontainer/devcontainer.json:16]: That's a good addition. I'll add it.

Comment (damienwebdev) [reply .devcontainer/devcontainer.json]: Done!

**Review by griest024 — DISMISSED** [review]
I was able to
- do a full build
- run a test in my local browser
- serve demo to my local browser
- run a test headlessly

I was on 4 cores and the build was noticeably slower but most of the other stuff was fast enough. Very cool though! LGTM

**Pull request merged**